File: 014/014.py
class Solution:
    def longestCommonPrefix(self, strs):

        ## Input:
        # strs: List[str]
        ## Output:
        # str

        min_strlen = 99999999
        for i in range(len(strs)):
            if (len(strs[i]) < min_strlen):
                min_strlen = len(strs[i])
        
        prefix = ""
        for i in range(min_strlen):
            ch = strs[0][i]
            for j in range(1, len(strs)):
                if (strs[j][i] != ch):
                    return prefix
            prefix += ch
        
        return prefix

File: 014/014.cpp
class Solution
{
public:
    string longestCommonPrefix(vector<string> &strs)
    {
        int min_strlen = 99999999;
        for (int i = 0; i < strs.size(); i++)
        {
            if (strs[i].length() < min_strlen)
            {
                min_strlen = strs[i].length();
            }
        }

        string prefix;
        prefix = "";
        for (int i = 0; i < min_strlen; i++)
        {
            char ch = strs[0][i];
            for (int j = 1; j < strs.size(); j++)
            {
                if (strs[j][i] != ch)
                {
                    return prefix;
                }
            }
            prefix += ch;
        }

        return prefix;
    }
};